modules/pages-and-forms/pages/widgets-containers.adoc

```diff
@@ -0,0 +1,79 @@
+
+=== Adding a row
+
+According your needs, a container can have any number of rows.
+
+To add a widget in a new row, a container has two dropzones where you can drop a widget from the palette. These dropzones are located at the top of the container and at the bottom of the container. When your mouse hovers the upper or lower border of a widget in a row, the dropzone becomes visible, as shown in these examples:
+
+Example 1: add a title widget in a new row above a paragraph widget:
+
+image::images/images-6_0/create-row-top.png[drop at the top]
+
+Example 2: add a paragraph widget in a new row below a title widget:
+
+image::images/images-6_0/create-row-bottom.png[drop at the bottom]
+
+When you drop a widget in one of these two zones, a new row is automatically created.
+
+=== Repeat content
+
+A container can repeat the content dynamically. This is useful if you need to repeat a set of components for each element in a data collection. Bind the *Repeat content* property to an array variable. To specify the current element of the collection, use `$item`. +
+This is useful if you want to display some property of the current element.
+
+=== Save as fragment
+
+You can save any container as a fragment (from the properties panel header, widget actions menu). This enables you to reuse the set of widgets in other pages.
+
+=== Tab container
+
+Use the *tab container* to structure a page using tabs.
+
+=== Tab container properties
+
+Tabs can be display vertically or horizontally. By default, tabs is stacked above the content. If you want to display tabs in the left of the content, you can follow this xref:ROOT:uid-vertical-tabs-container-tutorial.adoc[tutorial].
+
+Type property allows you to customize the tabs style ('tabs' or 'pills').
+
+=== Tab properties
+
+Each tab is a container widget. To edit a tab name, select the corresponding tab to display its properties.
+
+The tab's properties allows you to customize default display and behaviour of tabs based on data and/or JS expressions. For example you can hide or disable tab with a dynamic expression.
+
+Tab title can be calculated dynamically to display and *render html*.
+
+=== Form container
+
+Use the *form container* to enable form validation for input widgets. The form container also exposes a local *$form* which is the AngularJS form object associated to the form container. In the Bonita Studio, if you define a contract for the task or process and click on the UI Designer icon, the generated form already contains a form container to embed default input widgets to collect the information that need to be passed to the contract to be verified. You can then define the validation you need on the input. +
+For example, you can bind the button's disabled property to _$form.$invalid_ to prevent user from clicking the button, until the form inputs are valid.
+
+=== +++<a name="modal-container">++++++</a>+++ Modal container
```

Comment:
⚠️ please define a AsciiDoc id instead of using HTML syntax. This prevents portability.
id definition: see https://docs.asciidoctor.org/asciidoc/latest/attributes/id/#add-additional-anchors-to-a-section
Remember that an automatic anchor is also generated for all sections but its value depends on the section title (so it can changed when the titles changes)
- https://docs.asciidoctor.org/asciidoc/latest/sections/auto-ids/
- https://docs.asciidoctor.org/asciidoc/latest/macros/xref/#anchors

I guess that if you define an anchor here, this is to reference it in this page or in another page. The syntax for such reference is documented in https://docs.asciidoctor.org/asciidoc/latest/macros/xref/#internal-cross-references.
But I don't find any new content referencing the new anchor. Is it done on purpose or is it missing?